src/ui/base-page.js

```diff
@@ -0,0 +1,259 @@
+            },
+            {
+                record: {
+                    target: "{that}.model.locale",
+                    singleTransform: {
+                        type: "fluid.transforms.condition",
+                        condition: "{page}.model.uiLanguage",
+                        true: "{page}.model.uiLanguage",
+                        false: undefined
+                    },
+                    namespace: "uiLanguage"
+                },
+                target: "{that sjrk.storyTelling.templateManager}.options.modelRelay"
+            }
+        ],
+        events: {
+            onStoryListenToRequested: null,
+            onAllUiComponentsReady: {
+                events: {
+                    onMenuReady: "{menu}.events.onControlsBound",
+                    onUioReady: "onUioReady"
+                }
+            },
+            onPreferencesLoaded: null,
+            onPreferenceLoadFailed: null,
+            onContextChangeRequested: null, // this includes changes in visibility, language, etc.
+            onUioReady: null,
+            onUioPanelsUpdated: null
+        },
+        listeners: {
+            "onCreate.getStoredPreferences": {
+                funcName: "sjrk.storyTelling.base.page.getStoredPreferences",
+                args: ["{that}", "{cookieStore}"],
+                priority: "before:reloadUioMessages"
+            },
+            "onCreate.reloadUioMessages": {
+                func: "{that}.reloadUioMessages",
+                args: ["{that}.model.uiLanguage"]
+            },
+            "{menu}.events.onInterfaceLanguageChangeRequested": [{
+                func: "{that}.applier.change",
+                args: ["uiLanguage", "{arguments}.0.data"],
+                namespace: "changeUiLanguage"
+            },
+            {
+                func: "{that}.reloadUioMessages",
+                args: "{arguments}.0.data",
+                namespace: "reloadUioMessages"
+            },
+            {
+                func: "{that}.events.onContextChangeRequested.fire",
+                namespace: "onContextChangeRequested",
+                priority: "last"
+            }]
+        },
+        invokers: {
+            reloadUioMessages: {
+                funcName: "sjrk.storyTelling.base.page.reloadUioMessages",
+                args: [
+                    "{arguments}.0",
+                    "{uio}.prefsEditorLoader.messageLoader",
+                    "options.locale"
+                ]
+            }
+        },
+        modelListeners: {
+            uiLanguage: [{
+                funcName: "sjrk.storyTelling.base.page.renderAllUiTemplates",
+                args: ["{that}"],
+                namespace: "renderAllUiTemplates"
+            },
+            {
+                funcName: "fluid.set",
+                args: ["{uio}", "options.multilingualSettings.locale", "{change}.value"],
+                namespace: "updateUioLanguage"
+            }],
+            "*": {
```

Comment:
You probably mean "" here - the "*" syntax means that you want a separate call to set for every individual change to a top-level path - https://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#wildcards-in-model-path-references